src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -116,23 +118,80 @@ function getBlockDividerChunk(block: DraftBlockType, depth: number): Chunk {
   };
 }
 
-function getBlockTypeForTag(tag: string, lastList: ?string): DraftBlockType {
-  switch (tag) {
-    case 'h1':
-      return 'header-one';
-    case 'h2':
-      return 'header-two';
-    case 'li':
-      if (lastList === 'ol') {
-        return 'ordered-list-item';
-      }
-      return 'unordered-list-item';
-    case 'blockquote':
-      return 'blockquote';
-    case 'pre':
-      return 'code-block';
+function getListBlockType(
+  tag: string,
+  lastList: ?string
+): ?DraftBlockType {
+  if (tag === 'li') {
+    if (lastList === 'ol') {
+      return 'ordered-list-item';
+    }
+    return 'unordered-list-item';
+  }
+
+  return null;
+}
+
+function getBlockMapSupportedTags(
+  draftBlockRenderMap: DraftBlockRenderMap
+): Array<string> {
+  const blockTypes = Object.keys(draftBlockRenderMap);
+  const matchedTags = blockTypes.map(function(blockType) {
+    return draftBlockRenderMap[blockType].element;
+  }).sort();
+
+  const supportedTags = matchedTags.filter(function(tag, index) {
+    // we do not want unstyled to be viewed as a valid match
+    const notUnstyledTag = tag !== draftBlockRenderMap.unstyled.element;
+
+    // this will guarantee no duplicate tags added
+    const uniqueTag = matchedTags.indexOf(tag) === index;
+
+    return notUnstyledTag && uniqueTag;
+  });
+
+  return supportedTags;
+}
+
+// custom element conversions
+function getMultiMatchedType(
+  tag: string,
+  lastList: ?string,
+  multiMatchExtractor: Array<Function>
+): ?DraftBlockType {
+  for (var i = 0; i < multiMatchExtractor.length; i++) {
+    const matchType = multiMatchExtractor[i](tag, lastList);
+
+    if (matchType) {
+      return matchType;
+    }
+  }
+
+  return null;
+}
+
+function getBlockTypeForTag(
+  tag: string,
+  lastList: ?string,
+  draftBlockRenderMap: DraftBlockRenderMap
+): DraftBlockType {
+  const blockTypes = Object.keys(draftBlockRenderMap);
+  const matchedTypes = blockTypes.filter(function(blockType) {
+    return draftBlockRenderMap[blockType].element === tag || draftBlockRenderMap[blockType].wrapper === tag;
+  });
+  const matchesFound = matchedTypes.length;
+  const UNSTYLED = 'unstyled';
+
+  // if we dont have any matched type, return unstyled
+  // if we have one matched type return it
+  // if we have multi matched types use the multi-match function to determine which type is it
+  switch (matchesFound) {
+    case 0:
+      return UNSTYLED;
+    case 1:
+      return matchedTypes[0];
     default:
-      return 'unstyled';
+      return getMultiMatchedType(tag, lastList, [getListBlockType]) || UNSTYLED;
```

Comment:
In this iteration yes, I will include a new PR that allows to extend that. No reason to make this PR even larger. But currently thats the only core use case